akka-actor-typed/src/main/scala/akka/actor/typed/internal/receptionist/LocalReceptionist.scala

```diff
@@ -47,6 +49,12 @@ private[akka] object LocalReceptionist extends ReceptionistBehaviorProvider {
   final case class SubscriberTerminated[T](key: ServiceKey[T], ref: ActorRef[ReceptionistMessages.Listing[T]])
       extends InternalCommand
 
+  val terminationWatchers = new mutable.HashMap[String, ActorRef[Nothing]]();
+
+  def uniqueKey(key: ServiceKey[_], service: ActorRef[_]): String = {
+    key.toString + service.path.toStringWithoutAddress
```

Comment:
maybe create a `case class UniqueServiceKey(key: ServiceKey[_], service: ActorRef[_])` instead of using string concatenation?